satpy/readers/fci_l2_nc.py

```diff
@@ -133,18 +169,26 @@ def get_dataset(self, dataset_id, dataset_info):
             fill_value = dataset_info['fill_value']
         except KeyError:
             fill_value = np.NaN
-        float_variable = variable.where(variable != fill_value, mask_value).astype('float32', copy=False)
-        float_variable.attrs = variable.attrs
-        variable = float_variable
+
+        if dataset_info['file_type'] == 'nc_fci_test_clm':
+            data_values = variable.where(variable != fill_value, mask_value).astype('uint32', copy=False)
```

Comment:
This doesn't seem to be covered but looks quite critical. Should there be a test for it?

Reply:
test has been added to cover this line